src/frontend/src/optimizer/plan_node/generic/change_log.rs

```diff
@@ -58,32 +60,21 @@ impl<PlanRef: GenericPlanRef> GenericPlanNode for ChangedLog<PlanRef> {
                 "op",
             ));
         }
-        if self.need_changed_log_row_id {
+        if self.need_change_log_row_id {
             fields.push(Field::with_name(
                 risingwave_common::types::DataType::Serial,
-                "_changed_log_row_id",
+                "_change_log_row_id",
             ));
         }
         Schema::new(fields)
     }
 
     fn stream_key(&self) -> Option<Vec<usize>> {
-        match self.input.stream_key() {
-            Some(keys) => {
-                let mut keys = keys.to_vec();
-                if self.need_changed_log_row_id {
-                    keys.push(self.schema().len() - 1);
-                }
-                Some(keys)
-            }
-            None => {
-                if self.need_changed_log_row_id {
-                    let keys = vec![self.schema().len() - 1];
-                    Some(keys)
-                } else {
-                    None
-                }
-            }
+        if self.need_change_log_row_id {
+            let keys = vec![self.schema().len() - 1];
+            Some(keys)
+        } else {
+            None
```

Comment:
IIUC, `change_log`'s stream key must be not None and must be its row_id. In which case will we reach this branch?